uber/site_sections/registration.py

```diff
@@ -153,6 +153,65 @@ def change_badge(self, session, id, message='', **params):
             'attendee': attendee
         }
 
+    def qrcode_generator(self, data):
+        """
+        Takes a piece of data and encrypts it using the QR_CODE_PASSWORD
+        Then, adds the EVENT_QR_ID, and returns an Aztec barcode as an image stream.
+        Args:
+            data: A string to encrypt.
+
+        Returns: A PNG buffer. Use this function in an img tag's src='' to display an image.
+
+        """
+        if not c.QR_CODE_PASSWORD:
+            return False  # Don't generate a QR Code if we can't encrypt it
+
+        encrypted_data = c.EVENT_QR_ID + bytes.decode(binascii.hexlify(simplecrypt.encrypt(c.QR_CODE_PASSWORD, data)))
+
+        cherrypy.response.headers['Content-Type'] = "image/png"
+
+        checkin_barcode = treepoem.generate_barcode(
+            barcode_type='azteccode',
+            data=encrypted_data,
+            options={},
+        )
+        buffer = BytesIO()
+        checkin_barcode.save(buffer, "PNG")
+        buffer.seek(0)
+        return cherrypy.lib.file_generator(buffer)
```

Comment:
Yeah, testing it would probably be a good idea, though I'm not opposed to offloading this part elsewhere.